### PR TITLE
NRL-758 parameterise terraform

### PR DESCRIPTION
**Pull request by axelkrastek1-nhs**

### CODE REVIEW

File: terraform/infrastructure/ephemeral-resources.tf [hunk 1]
```diff
@@ -1,41 +1,17 @@
...
-  versioning_configuration {
-    status = "Enabled"
-  }
+module "ephemeral-pointers-table" {
+  count       = local.use_shared_resources ? 0 : 1
+  source      = "./modules/pointers-table"
+  name_prefix = local.prefix
 }
 
 resource "aws_s3_bucket_policy" "allow-lambda-to-read" {
```
Comment (axelkrastek1-nhs) [inline terraform/infrastructure/ephemeral-resources.tf]: Move to lambda.tf

File: tests/features/environment.py [hunk 2]
```diff
@@ -18,7 +18,8 @@ def before_all(context: Context):
     context.env = context.config.userdata.get("env")
     context.base_url = f"https://{context.env}.api.record-locator.dev.national.nhs.uk/"
 
+    default_table_name = f"nhsd-nrlf--{context.env}-pointers-table"
+    os.environ.setdefault("TABLE_NAME", default_table_name)
```
Comment (axelkrastek1-nhs) [inline tests/features/environment.py]: Remove this

**Review by mattdean3-nhs — APPROVED** [review]
LGTM - works a treat for me :heavy_check_mark:

**Review by mattdean3-nhs — APPROVED** [review]
LGTM

**Review by mattdean3-nhs — APPROVED** [review]
LGTM

**Pull request merged**